### PR TITLE
Add username and password section

### PR DESCRIPTION
**Pull request by Jessie212**
## General information

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Documentation only.

> How would you describe this change to a non-technical end user or system administrator?

Add username and password section

## Documentation

( ) No documentation is needed.
( :white_check_mark:) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( :white_check_mark:) No release notes entries required.
( ) Release notes entries required with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/installation/cli.rst [hunk 1]
```diff
@@ -46,17 +46,31 @@ The recommended TLS implementation is to use a globally trusted certificate. In
...
+The Trino CLI supports many authentication methods detailed in the following
+sections:
 
-Use ``--help`` to see information about specifying the keystore, truststore, and
-other authentication details as required. If using Kerberos, see :doc:`/security/cli`.
+Username and password authentication
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Usermane and password authentication is typically configured in a cluster with
+the ``PASSWORD`` authentication type. Password file authentication is very
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:62]: just link to the authentication type doc and get rid of the sentence "Password file ..."

File: docs/src/main/sphinx/installation/cli.rst [hunk 2]
```diff
@@ -46,17 +46,31 @@ The recommended TLS implementation is to use a globally trusted certificate. In
...
+similar to :doc:`/security/ldap` and :doc:`/security/salesforce`. You can
+override your username with the ``--user`` option.
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:64]: Before this sentence you need to say that the usename defaults to the operating system user. And if your clsuter needs a different username.. then you can use the override with --user

File: docs/src/main/sphinx/installation/cli.rst [hunk 3]
```diff
@@ -46,17 +46,31 @@ The recommended TLS implementation is to use a globally trusted certificate. In
...
+
+.. code-block:: text
+
+  trino --user=somebody@example.com
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:68]: use 

./trino 

since that is what all the other examples in this page do ..

Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:68]: we might want to send a separate PR that changes that after explaining that you need to add trino to the path..

File: docs/src/main/sphinx/installation/cli.rst [hunk 4]
```diff
@@ -46,17 +46,31 @@ The recommended TLS implementation is to use a globally trusted certificate. In
...
+
+Password file authentication defaults to your operating system username. If your
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:70]: Remove the "Password file" sentence.

File: docs/src/main/sphinx/installation/cli.rst [hunk 5]
```diff
@@ -116,6 +130,11 @@ JWT authentication
 To access a Trino cluster configured to use :doc:`/security/jwt`, use the
 ``--access-token=<token>`` option to pass a JWT to the server.
 
+Kerberos authentication
+^^^^^^^^^^^^^^^^^^^^^^^
+
+If you are using Kerberos, see :doc:`/security/cli`
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:136]: If your cluster is secured with Kerberos authenticatiomn , see LINK  for more details.

File: docs/src/main/sphinx/installation/cli.rst [hunk 6]
```diff
@@ -116,6 +130,11 @@ JWT authentication
 To access a Trino cluster configured to use :doc:`/security/jwt`, use the
 ``--access-token=<token>`` option to pass a JWT to the server.
 
+Kerberos authentication
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:133]: Note for other reviewers .. we are doing this for now in this PR.. we soon get rid of the other docs and migrate all the info into here.. but one step at a time

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/installation/cli.rst (hunk 2 above)
Comment (Jessie212) [reply docs/src/main/sphinx/installation/cli.rst:64]: I mistakenly chopped up the paragraph with the example.

File: docs/src/main/sphinx/installation/cli.rst [hunk 7]
```diff
@@ -46,17 +46,30 @@ The recommended TLS implementation is to use a globally trusted certificate. In
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Username and password authentication is typically configured in a cluster with
+the ``PASSWORD`` authentication type. It defaults to your operating system
+username. If your cluster uses a different username, you can override your
+username with the ``--user`` option.
+
+.. code-block:: text
+
+  ./trino --user=somebody@example.com
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:68]: add --server like in other snippet

File: docs/src/main/sphinx/installation/cli.rst [hunk 8]
```diff
@@ -46,17 +46,30 @@ The recommended TLS implementation is to use a globally trusted certificate. In
...
+
+If your Trino server requires password authentication, use the ``--password``
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:70]: Move this sentence above the code snippet and add --password to the example

File: docs/src/main/sphinx/installation/cli.rst [hunk 9]
```diff
@@ -262,3 +281,65 @@ To view debug information, including the stack trace for failures, use the
     ...
     at java.base/java.lang.Thread.run(Thread.java:834)
     select count(*) from tpch.tiny.nations
+
+============================================================== ======================================================================
```
Comment (mosabua) [inline docs/src/main/sphinx/installation/cli.rst:285]: Oh .. get rid of this in this PR .. add it to a separate PR please and use list-table syntax

Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst:285]: And add it in a separate section called


Options


with some intro sentence like

The following table lists all supported command line options for the CLI in alphabetical order.

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/installation/cli.rst [hunk 10]
```diff
@@ -46,17 +46,30 @@ The recommended TLS implementation is to use a globally trusted certificate. In
...
 
-Use ``--help`` to see information about specifying the keystore, truststore, and
-other authentication details as required. If using Kerberos, see :doc:`/security/cli`.
+Username and password authentication
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Username and password authentication is typically configured in a cluster with
+the ``PASSWORD`` authentication type. It defaults to your operating system
+username. If your cluster uses a different username, you can override your
+username with the ``--user`` option.
```
Comment (hashhar) [inline docs/src/main/sphinx/installation/cli.rst:64]: This part about username and it's default value isn't applicable only when password auth is enabled. So it should precede this section.

File: docs/src/main/sphinx/installation/cli.rst (hunk 6 above)
Comment (Ordinant) [reply docs/src/main/sphinx/installation/cli.rst:132]: 👣

File: docs/src/main/sphinx/installation/cli.rst (hunk 10 above)
Comment (mosabua) [reply docs/src/main/sphinx/installation/cli.rst]: Good point .. how about we leave this as is in this PR.. but move it into the table of options we are working on  in a follow up PR @hashhar  ..

Comment (hashhar) [reply docs/src/main/sphinx/installation/cli.rst]: Fine with me, can you link me to the other PR so that I remember to make sure this is applied?

Comment (Jessie212) [reply docs/src/main/sphinx/installation/cli.rst]: @hashhar https://github.com/trinodb/trino/pull/11187/files

**Review by mosabua — APPROVED** [review]

**Review by hashhar — APPROVED** [review]

**Pull request merged**